test: Add HTTP-CEL chainsaw test with test server

yrsuthari opened the pull request:
Add chainsaw conformance test for HTTP-CEL in ClusterPolicy validation with a dedicated test-server deployment. The test covers HTTP GET and POST methods, header validation, and response body handling.

## Explanation

This PR adds a new chainsaw conformance test for the HTTP-CEL feature in ClusterPolicy validation. The test validates that HTTP-CEL expressions can properly perform GET and POST requests, validate headers, and process response bodies through a dedicated test server deployment.

## Related issue

Closes #10544

## Milestone of this PR
<!--

Add the milestone label by commenting `/milestone 1.2.3`.

-->

## Documentation (required for features)

My PR contains new or altered behavior to Kyverno. 
- [ ] I have sent the draft PR to add or update [the documentation](https://github.com/kyverno/website) and the link is:
  <!-- Uncomment to link to the PR -->
  <!-- https://github.com/kyverno/website/pull/123 -->

## What type of PR is this

/kind test

## Proposed Changes

This PR adds a comprehensive test for HTTP-CEL functionality in ClusterPolicy validations. It includes:
- A dedicated test server deployment and service for HTTP endpoint testing
- A ClusterPolicy with HTTP-CEL expressions that test GET and POST methods
- Validation of header parameters and response body handling
- Proper assertions to verify the test server is working as expected

### Proof Manifests

```yaml
# Test Server Deployment and Service
apiVersion: apps/v1
kind: Deployment
metadata:
  name: test-server
  namespace: http-cel-test
spec:
  replicas: 1
  selector:
    matchLabels:
      app: test-server
  template:
    metadata:
      labels:
        app: test-server
    spec:
      containers:
      - name: test-server
        image: kyverno/test-server:latest
        ports:
        - containerPort: 8080
---
apiVersion: v1
kind: Service
metadata:
  name: test-server
  namespace: http-cel-test
spec:
  selector:
    app: test-server
  ports:
  - port: 8080
    targetPort: 8080
  type: ClusterIP
```

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md).
- [x] I have read the [PR documentation guide](https://github.com/kyverno/kyverno/blob/main/.github/pr_documentation.md) and followed the process including adding proof manifests to this PR.
- [ ] This is a bug fix and I have added unit tests that prove my fix is effective.
- [x] This is a feature and I have added CLI tests that are applicable.
- [ ] My PR needs to be cherry picked to a specific release branch which is <replace>.
- [ ] My PR contains new or altered behavior to Kyverno and
  - [ ] CLI support should be added and my PR doesn't contain that functionality.

yrsuthari commented:
@JimBugwadia PTAL. Thanks!

fjogeleit commented:
Hey, how does your PR relate to https://github.com/kyverno/kyverno/pull/10544?

If this PR relates to https://github.com/kyverno/kyverno/issues/12690 (which you mentioned in the readme) some clarification:

1. Our HTTP CEL lib is not available for the Policy and ClusterPolicy type. Only for our new ImageValidatingPolicy and ValidatingPolicy types
2. The Signature of our HTTP CEL lib is different to the one you used in your expression.

You can checkout https://github.com/kyverno/kyverno/tree/main/test/conformance/chainsaw/validating-policies/context/api-call as reference.

yrsuthari commented:
> Hey, how does your PR relate to #10544?
> 
> If this PR relates to #12690 (which you mentioned in the readme) some clarification:
> 
>     1. Our HTTP CEL lib is not available for the Policy and ClusterPolicy type. Only for our new ImageValidatingPolicy and ValidatingPolicy types
> 
>     2. The Signature of our HTTP CEL lib is different to the one you used in your expression.
> 
> 
> You can checkout https://github.com/kyverno/kyverno/tree/main/test/conformance/chainsaw/validating-policies/context/api-call as reference.

@fjogeleit Thank you for the feedback on the HTTP-CEL tests! I've addressed all your concerns:
- Replaced ClusterPolicy with ValidatingPolicy as you correctly pointed out that HTTP CEL library is only available for ValidatingPolicy and ImageValidatingPolicy types

- Updated the HTTP CEL function signatures to match the current implementation:
    - Changed from previous format to the correct http.Get() and http.Post() style
    - Properly handling the response structure with status field
    - Correctly referencing body fields in validations

- Restructured the policy following the reference you provided:
    - Using variables section for HTTP call definitions
    - Using validations section for response validation

Tested the implementation by adapting to use httpbin.org pattern endpoints similar to the reference implementation

I've manually tested these changes and confirmed the HTTP CEL functionality works correctly with various scenarios (GET/POST, headers handling, response validation). The test passes successfully with the updated ValidatingPolicy format.
Could you please review the changes when you have a chance? Thanks again for your guidance!

fjogeleit commented:
@yrsuthari the check test command is failing, please check if your template paths are correct
